### ISSUE TITLE
prefer-stateless-function displayName false-positive

### ISSUE THREAD

**Issue by fatfisz** [yannickcr/eslint-plugin-react#615]
I have a component factory that returns something like this:
```js
return React.createClass({
  displayName: `Wrapper<${someName}>`,
  ...
});
```

Because I can't otherwise create a function with a computed name without some gymnastics, I think that `prefer-statelless-function` should not warn me about a component like this.

`hasOtherProperties` has a simple check for `displayName` - I think this property should be checked in a separate method with a bit more complicated logic, e.g. checking whether it is a simple string literal or not.

I can do the PR if it is ok.

**Comment by yannickcr** [comment]
You can use `displayName` the same way on a stateless function:

```js
function Foo() {
  return <span>Bar</span>;
}
Foo.displayName = `Wrapper<${someName}>`;
```

**Comment by fatfisz** [comment]
Oh... thanks a lot, it didn't occur to me somehow.
